diff -up xine-lib-1.1.18/src/dxr3/dxr3_decode_spu.c.no_compat_c xine-lib-1.1.18/src/dxr3/dxr3_decode_spu.c
--- xine-lib-1.1.18/src/dxr3/dxr3_decode_spu.c.no_compat_c	2010-02-12 14:39:52.000000000 -0600
+++ xine-lib-1.1.18/src/dxr3/dxr3_decode_spu.c	2010-02-24 08:40:51.854952448 -0600
@@ -60,7 +60,7 @@
 #include "video_out_dxr3.h"
 #include "dxr3.h"
 
-#include "compat.c"
+/* #include "compat.c" */
 
 #define MAX_SPU_STREAMS 32
 
diff -up xine-lib-1.1.18/src/dxr3/dxr3_decode_video.c.no_compat_c xine-lib-1.1.18/src/dxr3/dxr3_decode_video.c
--- xine-lib-1.1.18/src/dxr3/dxr3_decode_video.c.no_compat_c	2010-02-12 14:39:52.000000000 -0600
+++ xine-lib-1.1.18/src/dxr3/dxr3_decode_video.c	2010-02-24 08:40:08.285952849 -0600
@@ -49,7 +49,7 @@
 #include "video_out_dxr3.h"
 #include "dxr3.h"
 
-#include "compat.c"
+/* #include "compat.c" */
 
 /* once activated, we wait for this amount of missing pan&scan info
  * before disabling it again */
diff -up xine-lib-1.1.18/src/dxr3/video_out_dxr3.c.no_compat_c xine-lib-1.1.18/src/dxr3/video_out_dxr3.c
--- xine-lib-1.1.18/src/dxr3/video_out_dxr3.c.no_compat_c	2010-02-12 14:39:52.000000000 -0600
+++ xine-lib-1.1.18/src/dxr3/video_out_dxr3.c	2010-02-24 08:41:12.490952692 -0600
@@ -66,7 +66,7 @@
 #include "dxr3.h"
 #include "video_out_dxr3.h"
 
-#include "compat.c"
+/* #include "compat.c" */
 
 /* the amount of extra time we give the card for decoding */
 #define DECODE_PIPE_PREBUFFER 10000
