diff -up xine-lib-1.1.16/src/combined/ffmpeg/ff_audio_decoder.c.ffmpeg_api xine-lib-1.1.16/src/combined/ffmpeg/ff_audio_decoder.c
--- xine-lib-1.1.16/src/combined/ffmpeg/ff_audio_decoder.c.ffmpeg_api	2009-01-07 10:14:32.000000000 -0600
+++ xine-lib-1.1.16/src/combined/ffmpeg/ff_audio_decoder.c	2009-01-07 12:04:34.000000000 -0600
@@ -271,7 +271,11 @@ static void ff_audio_decode_data (audio_
        * bits/sample for some codecs (e.g. MS ADPCM) */
       this->audio_bits = 16;  
   
+#if LIBAVCODEC_VERSION_INT < ((52<<16)+(0<<8)+0)
       this->context->bits_per_sample = this->audio_bits;
+#else
+     this->context->bits_per_coded_sample = this->audio_bits;
+#endif
       this->context->sample_rate = this->audio_sample_rate;
       this->context->channels    = this->audio_channels;
       this->context->codec_id    = this->codec->id;
@@ -329,7 +333,11 @@ static void ff_audio_decode_data (audio_
                               &decode_buffer_size,
                               &this->buf[0],
                               this->size);
+#if LIBAVCODEC_VERSION_INT < ((52<<16)+(0<<8)+0)
 	this->audio_bits = this->context->bits_per_sample;
+#else
+	this->audio_bits = this->context->bits_per_coded_sample;
+#endif
 	this->audio_sample_rate = this->context->sample_rate;
 	this->audio_channels = this->context->channels;
 	if (!this->audio_bits || !this->audio_sample_rate || !this->audio_channels)
diff -up xine-lib-1.1.16/src/combined/ffmpeg/ff_video_decoder.c.ffmpeg_api xine-lib-1.1.16/src/combined/ffmpeg/ff_video_decoder.c
--- xine-lib-1.1.16/src/combined/ffmpeg/ff_video_decoder.c.ffmpeg_api	2009-01-07 10:14:32.000000000 -0600
+++ xine-lib-1.1.16/src/combined/ffmpeg/ff_video_decoder.c	2009-01-07 11:58:03.000000000 -0600
@@ -946,7 +946,11 @@ static void ff_handle_header_buffer (ff_
               this->context->extradata_size);
       }
       
+#if LIBAVCODEC_VERSION_INT < ((52<<16)+(0<<8)+0)
       this->context->bits_per_sample = this->bih.biBitCount;
+#else
+      this->context->bits_per_coded_sample = this->bih.biBitCount; 
+#endif
             
     } else {
     
