--- xine-lib-1.1.21/src/combined/ffmpeg/ffmpeg_decoder.h.orig	2013-08-28 22:43:33.653572011 +0200
+++ xine-lib-1.1.21/src/combined/ffmpeg/ffmpeg_decoder.h	2013-08-28 22:44:24.087787844 +0200
@@ -35,7 +35,7 @@
 
 typedef struct ff_codec_s {
   uint32_t          type;
-  enum CodecID      id;
+  enum AVCodecID      id;
   const char       *name;
 } ff_codec_t;
 
--- xine-lib-1.1.21/src/combined/ffmpeg/ff_audio_decoder.c.orig	2013-08-28 23:07:28.399833089 +0200
+++ xine-lib-1.1.21/src/combined/ffmpeg/ff_audio_decoder.c	2013-08-28 23:09:41.375423336 +0200
@@ -47,6 +47,11 @@
 
 #define AUDIOBUFSIZE (64 * 1024)
 
+#ifndef AVCODEC_MAX_AUDIO_FRAME_SIZE
+/* from libavcodec/avcodec.h dated Dec 23 2012 */
+#define AVCODEC_MAX_AUDIO_FRAME_SIZE 192000 // 1 second of 48khz 32bit audio
+#endif
+
 typedef struct {
   audio_decoder_class_t   decoder_class;
 } ff_audio_class_t;
--- xine-lib-1.1.21/src/combined/ffmpeg/ff_video_decoder.c.orig	2013-08-28 23:30:14.261671159 +0200
+++ xine-lib-1.1.21/src/combined/ffmpeg/ff_video_decoder.c	2013-08-28 23:30:55.725849741 +0200
@@ -1055,7 +1055,7 @@
         this->bih.biWidth  = _X_BE_16(&this->buf[12]);
         this->bih.biHeight = _X_BE_16(&this->buf[14]);
 
-        this->context->sub_id = _X_BE_32(&this->buf[30]);
+/*        this->context->sub_id = _X_BE_32(&this->buf[30]); */
 
         this->context->slice_offset = calloc(SLICE_OFFSET_SIZE, sizeof(int));
         this->slice_offset_size = SLICE_OFFSET_SIZE;
