diff -up xine-lib-1.1.19/src/combined/ffmpeg/ff_audio_decoder.c.ffmpeg xine-lib-1.1.19/src/combined/ffmpeg/ff_audio_decoder.c
--- xine-lib-1.1.19/src/combined/ffmpeg/ff_audio_decoder.c.ffmpeg	2010-03-23 16:41:49.000000000 +0100
+++ xine-lib-1.1.19/src/combined/ffmpeg/ff_audio_decoder.c	2011-08-27 19:21:47.000000000 +0200
@@ -286,11 +286,20 @@ static void ff_audio_decode_data (audio_
     if (!this->output_open) {
       if (!this->audio_bits || !this->audio_sample_rate || !this->audio_channels) {
         decode_buffer_size = AVCODEC_MAX_AUDIO_FRAME_SIZE;
+#if FF_API_AUDIO_OLD
         avcodec_decode_audio2 (this->context,
                               (int16_t *)this->decode_buffer,
                               &decode_buffer_size,
                               &this->buf[0],
                               this->size);
+#else
+        AVPacket avpkt;
+        av_init_packet(&avpkt);
+        avpkt.data = &this->buf[0];
+        avpkt.size = this->size;
+        avcodec_decode_audio3 (this->context, (int16_t *)this->decode_buffer,
+                               &decode_buffer_size, &avpkt);
+#endif
 	this->audio_bits = this->context->bits_per_sample;
 	this->audio_sample_rate = this->context->sample_rate;
 	this->audio_channels = this->context->channels;
@@ -311,12 +320,22 @@ static void ff_audio_decode_data (audio_
       offset = 0;
       while (this->size>0) {
         decode_buffer_size = AVCODEC_MAX_AUDIO_FRAME_SIZE;
+#if FF_API_AUDIO_OLD
         bytes_consumed = avcodec_decode_audio2 (this->context,
                                                (int16_t *)this->decode_buffer,
                                                &decode_buffer_size,
                                                &this->buf[offset],
                                                this->size);
 
+#else
+        AVPacket avpkt;
+        av_init_packet(&avpkt);
+        avpkt.data = &this->buf[offset];
+        avpkt.size = this->size;
+        bytes_consumed = avcodec_decode_audio3 (this->context,
+                                               (int16_t *)this->decode_buffer,
+                                               &decode_buffer_size, &avpkt);
+#endif
         if (bytes_consumed<0) {
           xprintf (this->stream->xine, XINE_VERBOSITY_DEBUG,
                    "ffmpeg_audio_dec: error decompressing audio frame\n");
diff -up xine-lib-1.1.19/src/combined/ffmpeg/ff_video_decoder.c.ffmpeg xine-lib-1.1.19/src/combined/ffmpeg/ff_video_decoder.c
--- xine-lib-1.1.19/src/combined/ffmpeg/ff_video_decoder.c.ffmpeg	2010-03-10 20:07:15.000000000 +0100
+++ xine-lib-1.1.19/src/combined/ffmpeg/ff_video_decoder.c	2011-08-27 20:07:35.000000000 +0200
@@ -1055,12 +1055,25 @@ static void ff_handle_mpeg12_buffer (ff_
     }
 
     /* skip decoding b frames if too late */
+#if FF_API_HURRY_UP
     this->context->hurry_up = (this->skipframes > 0);
+#else
+    this->context->skip_frame = (this->skipframes > 0);
+#endif
 
     lprintf("avcodec_decode_video: size=%d\n", this->mpeg_parser->buffer_size);
+#if FF_API_VIDEO_OLD
     len = avcodec_decode_video (this->context, this->av_frame,
                                 &got_picture, this->mpeg_parser->chunk_buffer,
                                 this->mpeg_parser->buffer_size);
+#else
+    AVPacket avpkt;
+    av_init_packet(&avpkt);
+    avpkt.data = (uint8_t *) this->mpeg_parser->chunk_buffer;
+    avpkt.size = this->mpeg_parser->buffer_size;
+    len = avcodec_decode_video2 (this->context, this->av_frame,
+                                 &got_picture, &avpkt);
+#endif
     lprintf("avcodec_decode_video: decoded_size=%d, got_picture=%d\n",
             len, got_picture);
     len = current - buf->content - offset;
@@ -1112,7 +1125,11 @@ static void ff_handle_mpeg12_buffer (ff_
 
     } else {
 
+#if FF_API_HURRY_UP
       if (this->context->hurry_up) {
+#else
+      if (this->context->skip_frame) {
+#endif
         /* skipped frame, output a bad frame */
         img = this->stream->video_out->get_frame (this->stream->video_out,
                                                   this->bih.biWidth,
@@ -1304,12 +1321,26 @@ static void ff_handle_buffer (ff_video_d
         got_picture = 0;
       } else {
         /* skip decoding b frames if too late */
+#if FF_API_HURRY_UP
         this->context->hurry_up = (this->skipframes > 0);
+#else
+        this->context->skip_frame = (this->skipframes > 0);
+#endif
 
         lprintf("buffer size: %d\n", this->size);
+#if FF_API_VIDEO_OLD
         len = avcodec_decode_video (this->context, this->av_frame,
                                     &got_picture, &chunk_buf[offset],
                                     this->size);
+#else
+        AVPacket avpkt;
+        av_init_packet(&avpkt);
+        avpkt.data = (uint8_t *) &chunk_buf[offset];
+        avpkt.size = this->size;
+        len = avcodec_decode_video2 (this->context, this->av_frame,
+                                     &got_picture, &avpkt);
+#endif
+
 
 #ifdef AVCODEC_HAS_REORDERED_OPAQUE
         /* reset consumed pts value */